src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/TelemetryInitializerBase.cs

```diff
@@ -33,15 +33,18 @@ public void Initialize(ITelemetry telemetry)
                 return;
             }
 
-            var request = context.Features.Get<RequestTelemetry>();
-
-            if (request == null)
+            lock (context)
```

Comment:
Failure to retrieve `RequestTelemetry` from `Features` is just a first symptom of parallel `HttpContext` access so fixing it would not really fix anything, context has multiple collections inside that are not thread safe.